verify/verify.go

```diff
@@ -83,3 +83,28 @@ func (verifier *ReproducibleProvenanceVerifier) Verify(provenanceFilePath string
 
 	return nil
 }
+
+// AmberProvenanceMetadataVerifier verifies Amber provenances by comparing the
+// content of the provenance predicate against a given set of expected values.
+type AmberProvenanceMetadataVerifier struct {
+	// TODO(#69): Add metadata fields.
+}
+
+// Verify verifies a given Amber provenance file by checking its content
+// against the expected values specified in this
+// AmberProvenanceMetadataVerifier instance. Returns an error if any of the
+// values is not as expected. Otherwise returns nil, indicating success.
+func (verifier *AmberProvenanceMetadataVerifier) Verify(provenanceFilePath string) error {
+	provenance, err := slsa.ParseProvenanceFile(provenanceFilePath)
+	if err != nil {
+		return fmt.Errorf("couldn't load the provenance file from %s: %v", provenanceFilePath, err)
+	}
+
+	if provenance.Predicate.BuildType != common.AmberBuildTypeV1 {
+		return fmt.Errorf("incorrect BuildType: got %s, want %v", provenance.Predicate.BuildType, common.AmberBuildTypeV1)
+	}
+
+	// TODO(#69): Check metadata against the expected values.
```

Comment:
Maybe we want to indicate the todo also in the documentation of the func?

Reply:
Thanks. Done.